pdm/project/core.py

```diff
@@ -581,6 +586,19 @@ def make_candidate_info_cache(self) -> CandidateInfoCache:
     def make_hash_cache(self) -> HashCache:
         return HashCache(directory=self.cache("hashes").as_posix())
 
+    def _yield_python_info_from_path(self, path: str | Path) -> Iterable[PythonInfo]:
+        """
+        Helper method to not yield corrupted python info.
+        """
+        try:
+            yield PythonInfo.from_path(path)
+        except InvalidPyVersion:
+            self.core.ui.echo(
+                f"This interpreter {str(path)!r} is corrupted, skip.",
+                fg="yellow",
+                err=True,
+            )
```

Comment:
Are we going to ignore all these errors? I am afraid we will fail to catch bugs and select the wrong Python interpreters in the end.

Reply:
It needs to refactor the class method `PythonInfo.from_path` and add an `invalid` attribute into `PythonInfo` class. So when a user selects a corrupted version, the `do_use` function can raise `InvalidPyVersion` error.

It sounds better way to resolve this problem.